### PR TITLE
Fixes Prettier to be integrated

### PR DESCRIPTION
**Pull request by ellenfang77**
We have created a workflow job that runs Prettier on all code changes before it is committed to the GitHub repository. 
This integration into the workflow helps to ensure that all code is formatted correctly and consistently before being merged into the main branch. Types of files to be formatted include: .json, js, .yaml, .md.

Prettier static analysis tool (https://prettier.io/): an opinionated code formatter that enforces a consistent style through parsing code and reprinting it based on its own rules. By leveraging a predefined set of rules, Prettier find a solution for all formatting situations, such as line breaks, spacing, indents, etc.

Installation Steps (https://prettier.io/docs/en/install):

- Run `npm install --save-dev --save-exact prettier` in the terminal
- Create an empty config file node --eval "fs.writeFileSync('.prettierrc','{}\n')"
- Create a .prettierignore file to let Prettier CLI know which files not to format (model after .gitignore)

Addressing compatibility with ESLint:

- Installs [https://github.com/prettier/eslint-config-prettier#installation](https://github.com/CMU-313/fall23-nodebb-formula1/pull/url) to make eslint and prettier compatible `npm install --save-dev eslint-config-prettier`
- Adds eslint-config-prettier to .eslintrc file: add "prettier" to the "extends" array
- Adds files that should be [ignored](https://prettier.io/docs/en/ignore.html) by prettier to .prettierignore (ESLint and Prettier face conflicting rules in areas such as concatenation, indentation, and line breaks.)
- Adds [configuraton](https://prettier.io/docs/en/configuration.html) to .prettierrc (The tool will use this configuration file to format code when it is run. )

Usage Steps:

- To format all files with Prettier, run `npx prettier . --write`
- To format a certain directory or file, run `prettier --write <name>`
- To check the format of files, run `prettier --check <name>`

Changes:

- Integrates Prettier tool into workflow
- Installs Prettier and eslint-config-prettier
- Adds empty config file .prettierrc to let other tools know we are using Prettier
- Adds file .prettierignore to let Prettier CLI and editors know what to not format
- Adds "prettier" to the "extends" array in .eslintrc

Artifacts:

Shortened terminal output when running npx prettier . --write in the terminal (actual output is too long)
image
Output when using Prettier on a specific directory npx prettier --write src/categories
image
Output when using Prettier on a specific file npx prettier --write public/language/ar/admin/menu.json
image
Output when running np
Screenshot 2023-11-02 at 5 05 14 PM

### CODE REVIEW

**Comment by cwesterink** [comment]
Could you add a list of file types that will be formatted with prettier? (e.g  .js, .ts, .yaml...)
Everything else looks good.

**Comment by Sebastian0011** [comment]
Do we ignore all files that Prettier could not format. If not, it would be nice to know which files (or file types) in particular are problematic

**Comment by Sebastian0011** [comment]
For future reference, it would probably be good to include the config documentation somewhere for Prettier. I think this is the source? https://prettier.io/docs/en/configuration.html

File: .github/workflows/prettier.yaml [hunk 1]
```diff
@@ -0,0 +1,83 @@
+name: Prettier Lint
+
+on:
+    push:
+        branches:
+            - main
```
Comment (Sebastian0011) [inline .github/workflows/prettier.yaml:6]: Does this only run when we push to main? I believe we also have this test run when we push to the branch as well.

Comment (ellenfang77) [reply .github/workflows/prettier.yaml:6]: It is the same for test and lint yaml files. Prettier Lint runs when pushed to the branch.

**Review by Sebastian0011 — APPROVED** [review]

**Pull request merged**